Electron/Core/RoveProtocol/Rovecomm.ts

```diff
@@ -7,49 +7,41 @@ const EventEmitter = require("events")
 function decodePacket(
   size: number,
   dataLength: number,
-  data: Uint8Array
+  data: Buffer
 ): number[] {
   const retArray = []
-  let i = 0
-  let j = 0
+  let readBytes
+
   switch (size) {
     case 1:
-      while (i < dataLength) {
-        retArray.push(Number(data[i]))
-        i += 1
+      readBytes = function (i: number) {
+        return data.readUInt8(i)
```

Comment:
I don't think using a readUint or readInt function here is the correct course of action, We have some packets which contain signed information versus some that contain unsigned information under last years standards iirc.